sympy/matrices/expressions/tests/test_permutation.py

```diff
@@ -0,0 +1,144 @@
+from sympy.combinatorics import Permutation, SymmetricGroup
+from sympy.core.expr import unchanged
+from sympy.matrices import Matrix
+from sympy.matrices.expressions import \
+    MatMul, BlockDiagMatrix, Determinant, Inverse
+from sympy.matrices.expressions.matexpr import \
+    MatrixSymbol, Identity, ZeroMatrix, OneMatrix
+from sympy.matrices.expressions.permutation import \
+    MatrixPermute, PermutationMatrix
+from sympy.utilities.pytest import raises
+from sympy import Symbol
+
+
+def test_PermutationMatrix_basic():
+    p = Permutation([1, 0])
+    assert unchanged(PermutationMatrix, p)
+    raises(ValueError, lambda: PermutationMatrix((0, 1, 2)))
+    assert PermutationMatrix(p).as_explicit() == Matrix([[0, 1], [1, 0]])
+    assert isinstance(PermutationMatrix(p) * MatrixSymbol('A', 2, 2), MatMul)
+
+
+def test_PermutationMatrix_matmul():
+    p = Permutation([1, 2, 0])
+    P = PermutationMatrix(p)
+    M = Matrix([[0, 1, 2], [3, 4, 5], [6, 7, 8]])
+    assert (P * M).as_explicit() == P.as_explicit() * M
+    assert (M * P).as_explicit() == M * P.as_explicit()
+
+    P1 = PermutationMatrix(Permutation([1, 2, 0]))
+    P2 = PermutationMatrix(Permutation([2, 1, 0]))
+    P3 = PermutationMatrix(Permutation([1, 0, 2]))
+    assert P1 * P2 == P3
+
+
+def test_PermutationMatrix_matpow():
+    p1 = Permutation([1, 2, 0])
+    P1 = PermutationMatrix(p1)
+    p2 = Permutation([2, 0, 1])
+    P2 = PermutationMatrix(p2)
+    assert P1**2 == P2
+    assert P1**3 == Identity(3)
+
+
+def test_PermutationMatrix_identity():
+    p = Permutation([0, 1])
+    assert PermutationMatrix(p).is_Identity
+
+    p = Permutation([1, 0])
+    assert not PermutationMatrix(p).is_Identity
+
+
+def test_PermutationMatrix_determinant():
+    for perm in SymmetricGroup(3).generate():
+        mat = PermutationMatrix(perm)
+        det = Determinant(mat).doit()
+        assert det == mat.as_explicit().det()
+
+
+def test_PermutationMatrix_inverse():
+    for perm in SymmetricGroup(3).generate():
+        mat = PermutationMatrix(perm)
+        inv = Inverse(mat).as_explicit()
+        assert inv == mat.as_explicit().inv()
+
+
+def test_PermutationMatrix_rewrite_BlockDiagMatrix():
+    for perm in SymmetricGroup(6).generate():
+        mat = PermutationMatrix(perm)
+        rewrite = mat.rewrite(BlockDiagMatrix)
+        assert mat.as_explicit() == rewrite.as_explicit()
```

Comment:
This test takes a lot of time because I have not provided limited set of examples.